hazelcast/src/main/java/com/hazelcast/query/PartitionPredicate.java

```diff
@@ -33,10 +35,26 @@
 public interface PartitionPredicate<K, V> extends Predicate<K, V> {
 
     /**
-     * Returns the partition key that determines the partition the {@linkplain
+     * Returns the partition keys that determine the partitions the {@linkplain
      * #getTarget() target} {@link Predicate} is going to execute on.
      *
-     * @return the partition key
+     * A default implementation of {@linkplain #getPartitionKeys() partition keys}
+     * that wraps the {@linkplain #getPartitionKey() partition key} in a singleton
+     * collection is provided for backwards compatibility.
+     *
+     * @return the partition keys
+     */
+    default Collection<? extends Object> getPartitionKeys() {
```

Comment:
Seems it is not required to have a default implementation for `getPartitionKeys`. Since in `PartitionPredicateImpl` we always override it.

Having only this should be enough:
```
    Collection<? extends Object> getPartitionKeys();
```